pulp_service/pulp_service/app/middleware.py

```diff
@@ -71,3 +74,9 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
             repository_name_var.set(repository_name)
         if 'HTTP_X_QUAY_AUTH' in request.META:
             x_quay_auth_var.set(request.META['HTTP_X_QUAY_AUTH'])
+
+
+class RhEdgeHostMiddleware(MiddlewareMixin):
+    def process_view(self, request, *args, **kwargs):
+        if "HTTP_X_RH_EDGE_HOST" in request.META and request.META["HTTP_X_RH_EDGE_HOST"] is not None:
+            request.META["HTTP_HOST"] = request.META["HTTP_X_RH_EDGE_HOST"]
```

Comment:
Are you planning to push again to replace `HTTP_HOST` with `HTTP_X_FORWARDED_HOST` also?

Reply:
Ok. I'm gonna work on it.